op-batcher/batcher/driver.go

```diff
@@ -115,10 +117,17 @@ func NewBatchSubmitter(ctx context.Context, cfg Config, l log.Logger, m metrics.
 
 	cfg.metr = m
 
+	syncStatus, err := fetchSyncStatus(ctx, cfg.RollupNode, cfg.NetworkTimeout)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
adding the safe-L2 block ref as constructor argument is probably better, so the caller can manage retries to deal with any network errors, before creating the batch-submitter object.

Reply:
I revisited this, and think we should remove the notion of `lastProcessedBlock` from the channel-manager entirely. The span-batch builder can tell from the L2 blocks themselves whether or not the L1 epoch changed in the first L2 block of the span. (by checking if the "seqnum" of the L1 block info of the block is non-zero).